PushPullLegs/View/Data Presentation/Workout Data/WorkoutDataViewController.swift

```diff
@@ -55,9 +56,17 @@ class WorkoutDataViewController: DatabaseTableViewController {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         guard let vm = workoutDataViewModel else { return }
         vm.selectedIndex = indexPath
-        let exerciseVm = ExerciseViewModel(exercise: vm.getSelected() as! Exercise)
+        var exerciseVm: ExerciseViewModel
+        var vc: DBExerciseViewController
+        guard let selected = vm.getSelected() as? NSManagedObject else { return }
```

Comment:
refactor this condition out into a factory pattern that returns either vc+vm pair